src/transformers/cache_utils.py

```diff
@@ -430,14 +558,18 @@ def update(
             self._seen_tokens += key_states.shape[-2]
 
         if len(self.key_cache) <= layer_idx:
-            self._quantized_key_cache.append(self._quantize(key_states.contiguous(), axis=self.axis_key))
-            self._quantized_value_cache.append(self._quantize(value_states.contiguous(), axis=self.axis_value))
+            self._quantized_key_cache.append(
+                self._quantize(key_states.contiguous(), axis=self.axis_key, cache_type="key")
+            )
+            self._quantized_value_cache.append(
+                self._quantize(value_states.contiguous(), axis=self.axis_value, cache_type="value")
+            )
             self.key_cache.append(torch.zeros(0, dtype=key_states.dtype, device=key_states.device))
             self.value_cache.append(torch.zeros(0, dtype=key_states.dtype, device=key_states.device))
             keys_to_return, values_to_return = key_states, value_states
         else:
-            dequant_key = self._dequantize(self._quantized_key_cache[layer_idx])
-            dequant_value = self._dequantize(self._quantized_value_cache[layer_idx])
+            dequant_key = self._dequantize(self._quantized_key_cache[layer_idx], cache_type="key")
```

Comment:
Instead of rewriting `update()` method for `CompressedTensorsCache` I decided to expand the interface.